lxd/migration/wsproto.go

```diff
@@ -0,0 +1,71 @@
+package migration
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/gorilla/websocket"
+
+	"github.com/lxc/lxd/shared"
+)
+
+// ProtoRecv gets a protobuf message from a websocket
+func ProtoRecv(ws *websocket.Conn, msg proto.Message) error {
+	mt, r, err := ws.NextReader()
+	if err != nil {
+		return err
+	}
+
+	if mt != websocket.BinaryMessage {
+		return fmt.Errorf("Only binary messages allowed")
+	}
+
+	buf, err := ioutil.ReadAll(r)
```

Comment:
Would we benefit from using our EAGAIN reader here as well?

Reply:
r over there is a gorilla websocket file which I don't believe has EAGAIN (it's not a kernel issued fd)